### PR TITLE
[ONNX] Export aten::mul(bool, bool) to onnx::and instead of onnx::mul

### PR DESCRIPTION
**Pull request by wschin**
As title. We don't need to implement `onnx::mul` for bool because it's already `onnx::and`. It's also semantically better to keep boolean to use their specific operations (AND, OR, XOR, etc). In addition, [ONNX](https://github.com/onnx/onnx/blob/main/docs/Operators.md#type-constraints-85) doesn't have `Mul` for boolean types.

### CODE REVIEW

**Comment by BowenBao** [comment]
Thank you for PR, could you validate if this test case works under your change?

```python
>>> a  = torch.tensor([True, False])
>>> b = torch.tensor([1., 2.])
>>> (a * b).dtype
torch.float32
```

**Comment by wschin** [comment]
> 

I don't think this will work.
```python
    def test_mul_float_bool(self):
        x = torch.tensor([True, False, True, False])
        y = torch.tensor([1.0, 2.0, 3.0, 4.0])
        self.assertONNX(lambda x, y: torch.mul(x, y), (x, y))
```
generates
```
ir_version: 4
producer_name: "pytorch"
producer_version: "1.11.0"
graph {
  node {
    input: "onnx::And_0"
    input: "onnx::And_1"
    output: "2"
    name: "And_0"
    op_type: "And"
  }
  name: "torch-jit-export"
  input {
    name: "onnx::And_0"
    type {
      tensor_type {
        elem_type: 9
        shape {
          dim {
            dim_value: 4
          }
        }
      }
    }
  }
  input {
    name: "onnx::And_1"
    type {
      tensor_type {
        elem_type: 1
        shape {
          dim {
            dim_value: 4
          }
        }
      }
    }
  }
  output {
    name: "2"
    type {
      tensor_type {
        elem_type: 9
        shape {
          dim {
            dim_value: 4
          }
        }
      }
    }
  }
}
opset_import {
  version: 9
}
```

**Comment by BowenBao** [comment]
@wschin will it work if you convert to `And` only when both self and other are bool type?

If so please add as test case under `test_pytorch_onnx_onnxruntime.py`.

File: torch/onnx/symbolic_helper.py [hunk 1]
```diff
@@ -314,16 +314,27 @@ def _slice_helper(g, input, axes, starts, ends, steps=None, dynamic_slice=False)
         from torch.onnx.symbolic_opset10 import _slice as _slice10
         return _slice10(g, input, axes, starts, ends, steps, dynamic_slice)
 
+_fp_tensor_types = set((torch.float16, torch.float32, torch.float64, torch.bfloat16))
```
Comment (garymm) [inline torch/onnx/symbolic_helper.py:317]: Let's add more structure:
```suggestion
_TypeGroup = collections.namedtuple("TypeGroup", ("tensor_types", "scalar_types"))
_FPTypeGroup = _TypeGroup((torch.float16, torch.float32, torch.float64, torch.bfloat16), ("Float", "Double", "Half", "BFloat16"))
_BoolTypeGroup = _TypeGroup((torch.bool,), ("Bool",))
```

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -314,16 +314,27 @@ def _slice_helper(g, input, axes, starts, ends, steps=None, dynamic_slice=False)
...
+_fp_scalar_types = set(("Float", "Double", "Half", "BFloat16"))
+_bool_tensor_types = set((torch.bool,))
+_bool_scalar_types = set(("Bool",))
+
+def _is_specific_type_group(value, tensor_types, scalar_types):
```
Comment (garymm) [inline torch/onnx/symbolic_helper.py:322]:
```suggestion
def _is_in_type_group(value: Union[torch.Tensor, torch._C.Value], type_group: _TypeGroup) -> bool:
```

**Review by garymm — CHANGES_REQUESTED** [review]
Please change the target branch from `master` to `onnx_ms_1`

File: torch/onnx/symbolic_helper.py (hunk 1 above)
Comment (wschin) [reply torch/onnx/symbolic_helper.py]: Sure.

File: torch/onnx/symbolic_helper.py (hunk 2 above)
Comment (wschin) [reply torch/onnx/symbolic_helper.py]: Looks great. Thanks.

**Comment by BowenBao** [comment]
@wschin please check CI ONNX test failure.

**Comment by BowenBao** [comment]
@wschin friendly ping. Please note you no longer need to change base from master to onnx_ms_1.

**Pull request closed without merging**